LogicalTypes.md

```diff
@@ -684,44 +689,58 @@ optional group my_list (LIST) {
 }
 ```
 
-Some existing data does not include the inner element layer. For
-backward-compatibility, the type of elements in `LIST`-annotated structures
+Some existing data does not include the inner element layer, resulting in a
+`LIST` that annotates a 2-level structure. Unlike the 3-level structure, the
+repetition of a 2-level structure can be `optional`, `required`, or `repeated`.
+When it is `repeated`, the `LIST`-annotated 2-level structure can only serve as
+an element within another `LIST`-annotated 2-level structure.
+
+For backward-compatibility, the type of elements in `LIST`-annotated structures
 should always be determined by the following rules:
 
 1. If the repeated field is not a group, then its type is the element type and
    elements are required.
 2. If the repeated field is a group with multiple fields, then its type is the
    element type and elements are required.
-3. If the repeated field is a group with one field and is named either `array`
+3. If the repeated field is a group with one field with `repeated` repetition,
+   then its type is the element type and elements are required.
+4. If the repeated field is a group with one field and is named either `array`
    or uses the `LIST`-annotated group's name with `_tuple` appended then the
    repeated type is the element type and elements are required.
-4. Otherwise, the repeated field's type is the element type with the repeated
+5. Otherwise, the repeated field's type is the element type with the repeated
```

Comment:
I have reverted most of the previous changes and now it should be clear. @etseidl @mapleFU 

To resolve a LIST-annotated group, we should apply rules in order:
- check if it is a 2-level structure (rule 1 to 3)
- check if it is a special 2-level structure (rule 4)
- it is a 3-level structure (rule 5)

Reply:
I don't want to add an example for rule 5 because it is already at line 685

Reply:
@pitrou Regarding to your [question](https://github.com/apache/parquet-format/pull/466#discussion_r1868965095) on rule 5, I think it is redundant to add another one.

Reply:
Redundant with what? We've been explicitly listing examples for all other rules.

Reply:
Ok, I added one with a slightly different schema. The inner field is `optional` in the new example. Please take a look.

Reply:
Ok, thanks.